### PR TITLE
Implement specialized tracking manager for e-/e+/gamma

### PR DESCRIPTION
**Pull request by hahnjo**
This PR implements a specialized tracking manager for e-/e+/gamma, making use of the new `G4VTrackingManager` interface in Geant4 11.0. The newly added EMMT physics list is around 5% faster than EMM for gg2ttbar events in Run3 geometry.

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

This is an upgrade for future, does not affect the current baseline

File: SimG4Core/PhysicsLists/src/CMSEmStandardPhysicsTrackingManager.cc [hunk 1]
```diff
@@ -0,0 +1,731 @@
+#include "G4Version.hh"
```
Comment (perrotta) [inline SimG4Core/PhysicsLists/src/CMSEmStandardPhysicsTrackingManager.cc:1]: This is redundant, given that it is already included in the header file

File: SimG4Core/PhysicsLists/src/CMSEmStandardPhysicsEMMT.cc [hunk 2]
```diff
@@ -0,0 +1,86 @@
+#include "G4Version.hh"
```
Comment (perrotta) [inline SimG4Core/PhysicsLists/src/CMSEmStandardPhysicsEMMT.cc:1]: This is redundant, given that it is already included in the header file

File: SimG4Core/PhysicsLists/src/CMSEmStandardPhysicsTrackingManager.cc [hunk 3]
```diff
@@ -0,0 +1,731 @@
...
+  fRangeFactor = p.getParameter<double>("G4MscRangeFactor");
+  fGeomFactor = p.getParameter<double>("G4MscGeomFactor");
+  fSafetyFactor = p.getParameter<double>("G4MscSafetyFactor");
+  fLambdaLimit = p.getParameter<double>("G4MscLambdaLimit") * CLHEP::mm;
+  std::string msc = p.getParameter<std::string>("G4MscStepLimit");
+  fStepLimitType = fUseSafety;
+  if (msc == "UseSafetyPlus") {
+    fStepLimitType = fUseSafetyPlus;
+  }
+  if (msc == "Minimal") {
```
Comment (perrotta) [inline SimG4Core/PhysicsLists/src/CMSEmStandardPhysicsTrackingManager.cc:64]: `else if`

**Review by perrotta — COMMENTED** [review]
A few very minor comments.
Depending on the urgency, they can be taken into account or not.

File: SimG4Core/PhysicsLists/src/CMSEmStandardPhysicsTrackingManager.cc (hunk 1 above)
Comment (hahnjo) [reply SimG4Core/PhysicsLists/src/CMSEmStandardPhysicsTrackingManager.cc:1]: Not quite, then I'd have to move the include of `SimG4Core/PhysicsLists/interface/CMSEmStandardPhysicsTrackingManager.h` before the version check. I can do that if you want, but I think it's generally recommended to include-what-you-use in every implementation file...

File: SimG4Core/PhysicsLists/src/CMSEmStandardPhysicsEMMT.cc (hunk 2 above)
Comment (hahnjo) [reply SimG4Core/PhysicsLists/src/CMSEmStandardPhysicsEMMT.cc:1]: Same as above

**Comment by hahnjo** [comment]
The latest push also includes `G4TrackingCut`, which I didn't have because the code was based on an older IB.

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**